dev/TeachingTip/TeachingTip.cpp

```diff
@@ -740,98 +808,105 @@ void TeachingTip::UpdateDynamicHeroContentPlacementToBottom()
 
 void TeachingTip::OnIsOpenChanged()
 {
-    m_acceleratorKeyActivatedRevoker = {};
-
-    if (IsOpen())
+    auto const strongThis = get_strong();
+    SharedHelpers::QueueCallbackForCompositionRendering([strongThis]() 
     {
-        //Reset the close reason to the default value of programmatic.
-        m_lastCloseReason = winrt::TeachingTipCloseReason::Programmatic;
+        if (strongThis->IsOpen())
+        {
```

Comment:
This change has a lot of cleanup stuff in it. Can you separate the cleanup from this bug fix so that I can cherry-pick the scoped bug fix into the release branch?